File: src/modifiers/allowsTightening.ts
import { createModifier } from './createModifier';

/// SwiftUI `.allowsTightening(_:)`. When `true`, the view is allowed to
/// compress inter-character spacing to fit text into the available space
/// before truncating. Defaults to `true`.
export function allowsTightening(value: boolean = true) {
  return createModifier('allowsTightening', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/AllowsTightening.swift
import SwiftUI

/// SwiftUI `.allowsTightening(_:)`. Lets the view compress inter-character
/// spacing to fit text before truncating. Registered in the generic
/// registry. Defaults to `true` when the value is absent.
enum RNWAllowsTighteningModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("allowsTightening") { view, params, _ in
            AnyView(view.allowsTightening(params.bool("value") ?? true))
        }
    }
}
